testpar/t_bigio.c

```diff
@@ -1876,9 +1881,37 @@ main(int argc, char **argv)
     if (newsize != oldsize)
         bigcount = newsize * 2;
 
-    MPI_Init(&argc, &argv);
-    MPI_Comm_size(MPI_COMM_WORLD, &mpi_size_g);
-    MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank_g);
+#ifdef H5_HAVE_TEST_API
+    /* Attempt to initialize with MPI_THREAD_MULTIPLE if possible */
+    if (MPI_SUCCESS != (mpi_code = MPI_Init_thread(&argc, &argv, required, &provided))) {
```

Comment:
If doing API testing, try to initialize parallel tests with maximum thread safety support since some VOL connectors like the Async and Cache VOL require this. Otherwise, just do a regular `MPI_Init` since `MPI_Init_thread` with `MPI_THREAD_MULTIPLE` can have performance consequences.